content/en/blog/2023/native-sidecars/index.md

```diff
@@ -0,0 +1,207 @@
+
+{{< text shell >}}
+$ cat <<EOF | kind create cluster --name sidecars --image gcr.io/istio-testing/kind-node:v1.28.0-beta.0 --config=-
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+featureGates:
+  SidecarContainers: true
+EOF
+{{< /text >}}
+
+Then we can download the latest Istio pre-release. I used Linux here; see [Dev Builds](https://github.com/istio/istio/wiki/dev-builds) for more steps.
+These images are built from the latest Istio commit, so again - do not try in prod!
+When we install, we will enable the feature flag for native sidecar support and turn on access logs to help demo things later.
+
+{{< text shell >}}
+$ TAG=$(curl https://storage.googleapis.com/istio-build/dev/latest)
+$ curl -L https://storage.googleapis.com/istio-build/dev/$TAG/istioctl-$TAG-linux-amd64.tar.gz | tar xz
+$ ./istioctl install --set values.pilot.env.ENABLE_NATIVE_SIDECARS=true -y --set meshConfig.accessLogFile=/dev/stdout
+{{< /text >}}
+
+And finally we can deploy a workload:
+
+{{< text shell >}}
+$ kubectl label namespace default istio-injection=enabled
+$ kubectl apply -f samples/sleep/sleep.yaml
+{{< /text >}}
+
+Let's look at the pod:
+
+{{< text shell >}}
+$ kubectl get pods
+NAME                     READY   STATUS    RESTARTS   AGE
+sleep-7656cf8794-8fhdk   2/2     Running   0          51s
+{{< /text >}}
+
+Everything looks normal at first glance...
+If we look under the hood, we can see the magic, though.
+
+{{< text shell >}}
+$ kubectl get pod -o "custom-columns="\
+"NAME:.metadata.name,"\
+"INIT:.spec.initContainers[*].name,"\
+"CONTAINERS:.spec.containers[*].name"
+
+NAME                     INIT                     CONTAINERS
+sleep-7656cf8794-8fhdk   istio-init,istio-proxy   sleep
+{{< /text >}}
+
+Here we can see all the `containers` and `initContainers` in the pod.
+
+Surprise! `istio-proxy` is now an `initContainer`.
+
+More specifically, it is an `initContainer` with `restartPolicy: Always` set (a new field, enabled by the `SidecarContainers` feature).
+This tells Kubernetes to treat it as a sidecar.
+
+This means that later containers in the list of `initContainers`, and all normal `containers` will not start until proxy container is first ready.
+Additionally, the pod will terminate even if the proxy container is still running.
+
+### Init container traffic
+
+To put this to the test, let's make our pod actually do something.
+Here we deploy a simple pod that sends a request in an `initContainer`.
+Normally, this would fail.
+
+{{< text yaml >}}
+apiVersion: v1
+kind: Pod
+metadata:
+  name: sleep
+spec:
+  initContainers:
+  - name: check-traffic
+    image: istio/base
+    command:
+    - curl
+    - httpbin.org/get
+  containers:
+  - name: sleep
+    image: istio/base
+    command: ["/bin/sleep", "infinity"]
+{{< /text >}}
+
+Checking the proxy container, we can see the request both succeeded and went through the Istio sidecar:
```

Comment:
Do we want to say something about how istio-proxy ends up injected before check-traffic? I assume if it doesn't, it would fail like before.

Reply:
Yes worth clarifying. I had the same q when i read it.  The sidecar should be injected & running (to my understanding) before checking the traffic, while previously without the new sidecar feature from k8s, the culr httpbin.org would likely run before istio-proxy reaches running thus failed.